clientv3/concurrency/election_test.go

```diff
@@ -64,14 +65,15 @@ func TestResumeElection(t *testing.T) {
 		string(leader.Kvs[0].Key), leader.Kvs[0].CreateRevision)
 
 	respChan := make(chan *clientv3.GetResponse)
+	errCHan := make(chan error)
 	go func() {
 		o := e.Observe(ctx)
 		respChan <- nil
 		for {
 			select {
 			case resp, ok := <-o:
 				if !ok {
-					t.Fatal("Observe() channel closed prematurely")
+					errCHan <- errors.New("observe() channel closed prematurely")
```

Comment:
cherry pick this to make code same as in 3.5 https://github.com/etcd-io/etcd/pull/14134/commits/31af5bdd96276beaab05751e944f98e1d2f677f4

Reply:
updated.